### ISSUE TITLE
[BUG] PHP Warning: Undefined array key "no_search_sub_entries" in typo3conf/ext/solr/Classes/IndexQueue/RecordMonitor.php line 203

### ISSUE THREAD

**Issue by LinkPool2** [TYPO3-Solr/ext-solr#3401]
**Describe the bug**
[BUG] PHP Warning: Undefined array key "no_search_sub_entries" in typo3conf/ext/solr/Classes/IndexQueue/RecordMonitor.php line 203

```
    protected function skipRecordByRootlineConfiguration(int $pid): bool
..
            if ($page['no_search_sub_entries']) {
..

```

should be?

```

    protected function skipRecordByRootlineConfiguration(int $pid): bool
    {
        /** @var RootlineUtility $rootlineUtility */
        $rootlineUtility = GeneralUtility::makeInstance(RootlineUtility::class, $pid);
        try {
            $rootline = $rootlineUtility->get();
        } catch (PageNotFoundException $e) {
            return true;
        }
        foreach ($rootline as $page) {
            if (isset($page['no_search_sub_entries']) && $page['no_search_sub_entries']) {
                return true;
            }
        }
        return false;
    }

```

**Comment by dkd-kaehm** [comment]
@LinkPool2 
Thanks for reporting that issue.
Could you please provide a pull request and port for that?


Current state:

https://github.com/TYPO3-Solr/ext-solr/blob/df336e11d0c01b550bca9ac7962e3e520e47e08c/Classes/IndexQueue/RecordMonitor.php#L193-L208

MUST be changed to `if ($page['no_search_sub_entries'] ?? false) { `:

```
protected function skipRecordByRootlineConfiguration(int $pid): bool 
 { 
     /** @var RootlineUtility $rootlineUtility */ 
     $rootlineUtility = GeneralUtility::makeInstance(RootlineUtility::class, $pid); 
     try { 
         $rootline = $rootlineUtility->get(); 
     } catch (PageNotFoundException $e) { 
         return true; 
     } 
     foreach ($rootline as $page) { 
         if ($page['no_search_sub_entries'] ?? false) { 
             return true; 
         } 
     } 
     return false; 
 } 
```

Target versions:

* release-11.5.x
* main

**Comment by dkd-kaehm** [comment]
Fixed in #3381, #3438
